doc/source/dev/toolchain.rst

```diff
@@ -0,0 +1,136 @@
+Toolchain and Roadmap
+=====================
+SciPy requires several tools in order to be built.
+
+- First and foremost, SciPy is a Python project hence it requires a Python environment.
+- The LAPACK and OpenBLAS numerical libraries need to be installed.
+- Compilers for C, Cython and Fortran code are needed.
```

Comment:
Even thought the C and C++ compilers are typically the same, let's be explicit, since C ≠ C++:
    
    Compilers for C, C++, Cython and Fortran code are needed.